### ISSUE TITLE
Reward doesn't go up

### ISSUE THREAD

**Issue by star277hk** [llSourcell/Reinforcement_Learning_for_Stock_Prediction#10]
After each episode, I expect agent learnt something from previous, but I plot the reward of 1000 Episode, the reward doesn't go up. Is it mean the agent can't learn from the dataset ?
I used EURUSD 2010 - 2015 total 1567 records with 3 window size
<img width="995" alt="screen shot 2018-07-31 at 3 04 05 pm" src="https://user-images.githubusercontent.com/18695558/43443440-735d339a-94d3-11e8-967f-53a89d7490cc.png">

**Comment by yeshengyi** [comment]
Same here, I think the training process should filter out those with less than expected profits.

**Comment by madytyoo** [comment]
I would suggest to adjust the normalization function, the implementation computes the sigmoid of returns:

`sigmoid(block[i + 1] - block[i])`

The returns in the sample stock file range between +/-100. For currencies the returns range is between  +/-0.001 if you are using a raw price data set. You could try to express the return in pips:

`sigmoid( (block[i + 1] - block[i])*10000)`

**Comment by MonaTanggg** [comment]
@madytyoo sorry tried, but the reward didn't go up after times 10000
![times1000](https://user-images.githubusercontent.com/25878147/43528745-3d44e6fe-95dc-11e8-8af8-f0bc7f929fc4.PNG)

**Comment by madytyoo** [comment]
@MonaTanggg I did a test with a EURUSD data  2009-2017, 1000 episodes logging the reward on Tensor board. I suggest to check your data set. 
![screenshot from 2018-08-02 00-51-00](https://user-images.githubusercontent.com/1027667/43553273-7433a03a-95ee-11e8-87d5-4e51e8f24fd2.png)

**Comment by star277hk** [comment]
@madytyoo did you edit anything from your source code/dataset , I wonder why I got a total different result, may I know your tensorflow version as well ? Not sure which part go wrong. Thanks in advance.

**Comment by madytyoo** [comment]
@star277hk  I did the changes suggested by xtr33me (see: https://github.com/llSourcell/Reinforcement_Learning_for_Stock_Prediction/pull/8).
In addition I multiplied the EURUSD data by 1000, in the `getStockDataVec` function.

`vec.append(float(line.split(",")[4])*1000)`

As I mentioned the currencies the returns range is between +/-0.001 is small to get a result. I run a test **without** multiplying the price by 1000 (see below)  and I got a result similar to your chart.
Cheers

![screenshot from 2018-08-02 11-29-15](https://user-images.githubusercontent.com/1027667/43575541-5f7fd484-9647-11e8-851b-c77e9c270c2f.png)

**Comment by star277hk** [comment]
@madytyoo yes I used xtr33me #8 added times 1000 at ` vec.append(float(line.split(",")[4])*1000)` still got no luck. I used EURUSD ASK 01.07.2010-01.07.2015 dataframe 1 day and window size 50 with tensorflow 1.3. May I know your setting ?

**Comment by madytyoo** [comment]
I'm using  EURUSD BID 01.01.2010-31.12.2017 1 day, window size 10, 1000 episodes, tensorflow 1.3.

**Comment by colbyham** [comment]
I added all columns of the dataset, high, low, volume, etc. to the input layer and i've been achieving max gain of ~60% over ten years for the GSPC dataset which is not that great but at least it's progress. Average drawdown of around 25-35%. I think adding actions to the agent that are related to how much of your portfolio you should hold, sell, or buy up to, etc would be helpful. In other words, bucketed ranges of sizes of trades based on portfolio and or buying power.

**Comment by MonaTanggg** [comment]
@madytyoo Strange, already using almost same setting but agent still can't learn. Are you just learning from the close price or added other values? If you don't mine, hope you can share me your project that can be raise reward, I wanna figure out where is the problem, thanks.
![test](https://user-images.githubusercontent.com/25878147/43676952-025ccc5c-982d-11e8-9c3d-b2ab4be6893d.PNG)

**Comment by xxdaggerxx** [comment]
kind off retarded to post code that doesn't work
